File: csharp_body_deref/Scripts/Map/MapScriptBaseObject.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Map.MapScriptBaseObject
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Map/MapScriptBaseObject.c
// Prior real C# source (older reference): Assets/Scripts/Map/MapScript/MapScriptBaseObject.cs
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Il2CppDummyDll;
using UnityEngine;
using Utility;

namespace Map;

[Token(Token = "0x2000277")]
public class MapScriptBaseObject : BaseCSVRow
{
	[Token(Token = "0x4000D6D")]
	[FieldOffset(Offset = "0x10")]
	[Order(1)]
	public string Type;

	[Token(Token = "0x4000D6E")]
	[FieldOffset(Offset = "0x18")]
	[Order(2)]
	public string Asset;

	[Token(Token = "0x4000D6F")]
	[FieldOffset(Offset = "0x20")]
	[Order(3)]
	public int Id;

	[Token(Token = "0x4000D70")]
	[FieldOffset(Offset = "0x24")]
	[Order(4)]
	public int Parent;

	[Token(Token = "0x4000D71")]
	[FieldOffset(Offset = "0x28")]
	[Order(5)]
	public bool Active;

	[Token(Token = "0x4000D72")]
	[FieldOffset(Offset = "0x29")]
	[Order(6)]
	public bool Static;

	[Token(Token = "0x4000D73")]
	[FieldOffset(Offset = "0x2A")]
	[Order(7)]
	public bool Visible;

	[Token(Token = "0x4000D74")]
	[FieldOffset(Offset = "0x2B")]
	[Order(8)]
	public bool Networked;

	[Token(Token = "0x4000D75")]
	[FieldOffset(Offset = "0x30")]
	[Order(9)]
	public string Name;

	[Token(Token = "0x4000D76")]
	[FieldOffset(Offset = "0x38")]
	[Order(10)]
	public float PositionX;

	[Token(Token = "0x4000D77")]
	[FieldOffset(Offset = "0x3C")]
	[Order(11)]
	public float PositionY;

	[Token(Token = "0x4000D78")]
	[FieldOffset(Offset = "0x40")]
	[Order(12)]
	public float PositionZ;

	[Token(Token = "0x4000D79")]
	[FieldOffset(Offset = "0x44")]
	[Order(13)]
	public float RotationX;

	[Token(Token = "0x4000D7A")]
	[FieldOffset(Offset = "0x48")]
	[Order(14)]
	public float RotationY;

	[Token(Token = "0x4000D7B")]
	[FieldOffset(Offset = "0x4C")]
	[Order(15)]
	public float RotationZ;

	[Token(Token = "0x4000D7C")]
	[FieldOffset(Offset = "0x50")]
	[Order(16)]
	public float ScaleX;

	[Token(Token = "0x4000D7D")]
	[FieldOffset(Offset = "0x54")]
	[Order(17)]
	public float ScaleY;

	[Token(Token = "0x4000D7E")]
	[FieldOffset(Offset = "0x58")]
	[Order(18)]
	public float ScaleZ;

	[Token(Token = "0x6000F3D")]
	[Address(RVA = "0x3FB4C10", Offset = "0x3FB4C10", VA = "0x3FB4C10")]
	public void SetPosition(Vector3 position)
	{
	}

	[Token(Token = "0x6000F3E")]
	[Address(RVA = "0x3FB4BF0", Offset = "0x3FB4BF0", VA = "0x3FB4BF0")]
	public void SetScale(Vector3 scale)
	{
	}

	[Token(Token = "0x6000F3F")]
	[Address(RVA = "0x3FB5C90", Offset = "0x3FB5C90", VA = "0x3FB5C90")]
	public void SetRotation(Vector3 rotation)
	{
	}

	[Token(Token = "0x6000F40")]
	[Address(RVA = "0x3FB4CE0", Offset = "0x3FB4CE0", VA = "0x3FB4CE0")]
	public void SetRotation(Quaternion rotation)
	{
	}

	[Token(Token = "0x6000F41")]
	[Address(RVA = "0x3FB5CB0", Offset = "0x3FB5CB0", VA = "0x3FB5CB0")]
	public Vector3 GetPosition()
	{
		return default(Vector3);
	}

	[Token(Token = "0x6000F42")]
	[Address(RVA = "0x3FB5CC0", Offset = "0x3FB5CC0", VA = "0x3FB5CC0")]
	public Vector3 GetRotation()
	{
		return default(Vector3);
	}

	[Token(Token = "0x6000F43")]
	[Address(RVA = "0x3FB5CD0", Offset = "0x3FB5CD0", VA = "0x3FB5CD0")]
	public Vector3 GetScale()
	{
		return default(Vector3);
	}

	[Token(Token = "0x6000F44")]
	[Address(RVA = "0x3FB5CE0", Offset = "0x3FB5CE0", VA = "0x3FB5CE0")]
	public MapScriptBaseObject()
	{
	}
}

File: cpp/Map/MapScriptBaseObject.c
// Type: Map.MapScriptBaseObject
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/Map/MapScriptBaseObject.cs
// Prior real C# source (older reference): Assets/Scripts/Map/MapScript/MapScriptBaseObject.cs
// --------------------------------

// Map.MapScriptBaseObject$$SetPosition
// il2cpp: void Map_MapScriptBaseObject__SetPosition (Map_MapScriptBaseObject_o* __this, UnityEngine_Vector3_o position, const MethodInfo* method);
// 0x40b4c10

void Map_MapScriptBaseObject__SetPosition
               (Map_MapScriptBaseObject_o *__this,UnityEngine_Vector3_o position,MethodInfo *method)

{
  *(float *)&(__this->fields).Name = position.fields.x;
  *(float *)((long)&(__this->fields).Name + 4) = position.fields.y;
  (__this->fields).PositionX = position.fields.z;
  return;
}


// Map.MapScriptBaseObject$$SetScale
// il2cpp: void Map_MapScriptBaseObject__SetScale (Map_MapScriptBaseObject_o* __this, UnityEngine_Vector3_o scale, const MethodInfo* method);
// 0x40b4bf0

void Map_MapScriptBaseObject__SetScale
               (Map_MapScriptBaseObject_o *__this,UnityEngine_Vector3_o scale,MethodInfo *method)

{
  (__this->fields).RotationY = scale.fields.x;
  (__this->fields).RotationZ = scale.fields.y;
  (__this->fields).ScaleX = scale.fields.z;
  return;
}


// Map.MapScriptBaseObject$$SetRotation
// il2cpp: void Map_MapScriptBaseObject__SetRotation (Map_MapScriptBaseObject_o* __this, UnityEngine_Vector3_o rotation, const MethodInfo* method);
// 0x40b5c90

void Map_MapScriptBaseObject__SetRotation
               (Map_MapScriptBaseObject_o *__this,UnityEngine_Vector3_o rotation,MethodInfo *method)

{
  (__this->fields).PositionY = rotation.fields.x;
  (__this->fields).PositionZ = rotation.fields.y;
  (__this->fields).RotationX = rotation.fields.z;
  return;
}


// Map.MapScriptBaseObject$$SetRotation
// il2cpp: void Map_MapScriptBaseObject__SetRotation (Map_MapScriptBaseObject_o* __this, UnityEngine_Quaternion_o rotation, const MethodInfo* method);
// 0x40b4ce0

void Map_MapScriptBaseObject__SetRotation_3fb4ce0
               (Map_MapScriptBaseObject_o *__this,UnityEngine_Quaternion_o rotation,MethodInfo *method)

{
  UnityEngine_Vector3_o UVar1;
  UnityEngine_Vector3_o euler;
  
  UVar1 = UnityEngine_Quaternion__Internal_ToEulerRad(rotation,(MethodInfo *)0x0);
  euler.fields.x = UVar1.fields.x * 57.29578;
  euler.fields.y = UVar1.fields.y * 57.29578;
  euler.fields.z = UVar1.fields.z * 57.29578;
  UVar1 = UnityEngine_Quaternion__Internal_MakePositive(euler,(MethodInfo *)0x0);
  (__this->fields).PositionY = UVar1.fields.x;
  (__this->fields).PositionZ = UVar1.fields.y;
  (__this->fields).RotationX = UVar1.fields.z;
  return;
}


// Map.MapScriptBaseObject$$GetPosition
// il2cpp: UnityEngine_Vector3_o Map_MapScriptBaseObject__GetPosition (Map_MapScriptBaseObject_o* __this, const MethodInfo* method);
// 0x40b5cb0

UnityEngine_Vector3_o
Map_MapScriptBaseObject__GetPosition(Map_MapScriptBaseObject_o *__this,MethodInfo *method)

{
  return (UnityEngine_Vector3_o)*(UnityEngine_Vector3_Fields *)&(__this->fields).Name;
}


// Map.MapScriptBaseObject$$GetRotation
// il2cpp: UnityEngine_Vector3_o Map_MapScriptBaseObject__GetRotation (Map_MapScriptBaseObject_o* __this, const MethodInfo* method);
// 0x40b5cc0

UnityEngine_Vector3_o
Map_MapScriptBaseObject__GetRotation(Map_MapScriptBaseObject_o *__this,MethodInfo *method)

{
  UnityEngine_Vector3_o UVar1;
  
  UVar1.fields.x = (__this->fields).PositionY;
  UVar1.fields.y = (__this->fields).PositionZ;
  UVar1.fields.z = (__this->fields).RotationX;
  return (UnityEngine_Vector3_o)UVar1.fields;
}


// Map.MapScriptBaseObject$$GetScale
// il2cpp: UnityEngine_Vector3_o Map_MapScriptBaseObject__GetScale (Map_MapScriptBaseObject_o* __this, const MethodInfo* method);
// 0x40b5cd0

UnityEngine_Vector3_o Map_MapScriptBaseObject__GetScale(Map_MapScriptBaseObject_o *__this,MethodInfo *method)

{
  UnityEngine_Vector3_o UVar1;
  
  UVar1.fields.x = (__this->fields).RotationY;
  UVar1.fields.y = (__this->fields).RotationZ;
  UVar1.fields.z = (__this->fields).ScaleX;
  return (UnityEngine_Vector3_o)UVar1.fields;
}


// Map.MapScriptBaseObject$$.ctor
// il2cpp: void Map_MapScriptBaseObject___ctor (Map_MapScriptBaseObject_o* __this, const MethodInfo* method);
// 0x40b5ce0

void Map_MapScriptBaseObject___ctor(Map_MapScriptBaseObject_o *__this,MethodInfo *method)

{
  undefined4 uVar1;
  
  if (g_data_057ac4e4 == '\0') {
    il2cpp_runtime_helper_023445d0(&"None");
    il2cpp_runtime_helper_023445d0(&"Unnamed");
    g_data_057ac4e4 = '\x01';
  }
  (__this->fields).Asset = "None";
  il2cpp_runtime_helper_022b4080(&(__this->fields).Asset);
  *(undefined2 *)&(__this->fields).Active = 0x101;
  *(undefined1 *)((long)&(__this->fields).Active + 2) = 1;
  uVar1 = "Unnamed"._4_4_;
  (__this->fields).Visible = (undefined4)"Unnamed";
  (__this->fields).Networked = uVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Visible);
  (__this->fields).RotationY = 1.0;
  (__this->fields).RotationZ = 1.0;
  (__this->fields).ScaleX = 1.0;
  Utility_BaseCSVRow___ctor((Utility_BaseCSVRow_o *)__this,(MethodInfo *)0x0);
  return;
}


